index.d.ts

```diff
@@ -0,0 +1,182 @@
+import { Context } from 'egg';
+
+export namespace levels {
+  export const ALL: number;
+  export const DEBUG: number;
+  export const INFO: number;
+  export const WARN: number;
+  export const ERROR: number;
+  export const NONE: number;
+}
+
+export type LoggerLevel = keyof typeof levels;
+
+export interface LoggerOptions {
+  level?: LoggerLevel;
+  encoding?: string;
+  consoleLevel?: LoggerLevel;
+}
+
+export interface EggLoggerOptions extends LoggerOptions {
+  file: string;
+  formatter?: any;
+  jsonFile?: string;
+  outputJSON?: boolean;
+  buffer?: boolean;
+  eol?: string;
+}
+
+export class Logger<T extends LoggerOptions = LoggerOptions> {
+  constructor(options: T);
+
+  /**
+   * disable a transport
+   * @param {String} name - transport name
+   */
+  disable(name: string): void;
+
+  /**
+   * enable a transport
+   * @param {String} name - transport name
+   */
+  enable(name: string): void;
+
+  /**
+   * Send log to all transports.
+   * It's proxy to {@link Transport}'s log method.'
+   * @param {String} level - log level
+   * @param {Array} args - log arguments
+   * @param {Object} meta - log meta
+   */
+  log(level: string, args: any[], meta: object): void;
```

Comment:
`args` is optional?

Reply:
no, see https://github.com/eggjs/egg-logger/blob/master/lib/logger.js#L64